### PR TITLE
Setup Association Relation

### PR DESCRIPTION
**Pull request by PrangonGhose**
### Changes made in this PR
- Created `Book`, `Classroom`, `Rental` class.
- Implement _has-a_ and _belongs-to_ relationship between `Classroom` and `Student` class.
- Implement _has-a_ and _belongs-to_ relationship between `Book`, `Rental` and `Person` class.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (MeqdamAlqudah) [inline book.rb]: - [ ] Please   Getters and Setters for @rentals should be present, add rentals to attr_accessor.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rentals(rental)
+    @rentals.push(rental)
+    rental.book = self
+    rental.person.add_rentals(rentals) unless rental.person.rentals.include?(rental)
+  end
```
Comment (MeqdamAlqudah) [inline book.rb]: - [ ] Please `the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.` the method should get the person and the data and then it should create a new rental object with the date ,person and self which is the book info.

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
```
Comment (MeqdamAlqudah) [inline rental.rb]: - [ ] Please  the rental should add itself to the @book achieving the 'belongs-to' association` 
it should look something like this 
@book.rentals  << self 
- [ ] Please  the rental should add itself to the @person achieving the 'belongs-to' association` 
it should look something like this 
@person.rentals  << self

File: person_class.rb [hunk 4]
```diff
@@ -7,14 +7,15 @@ def initialize(age, name = 'Unknown', parent_permission: true)
...
+    @rentals = []
     super()
   end
 
   # defining accessor get and set method
   attr_accessor :name, :age
 
   # defining accessor get method
-  attr_reader :id
+  attr_reader :id, :rentals, :parent_permission
```
Comment (MeqdamAlqudah) [inline person_class.rb]: - [ ] Please getters and Setters for @rentals should be present

**Review by MeqdamAlqudah — CHANGES_REQUESTED** [review]
Hi @PrangonGhose 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image


## Highlights
:+1: no linter errors 
:+1: file structure

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PrangonGhose** [comment]
### Changes made after first review
- Fix setters and getters in `Classroom`, `Book`, and `Person` class.
- Fix _has-many_ association in `Classroom`, `Student`, `Book`, `Rentals` and `Person` class.
- Fix `Rental` class to add itself in corresponding `Book` and `Person` classes to achieve _belongs-to_ association.

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @PrangonGhose, 

To highlight 👍🏽 
✔️ No linter errors
✔️ All associations are set

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**